### PR TITLE
fix(Core/ObjectMgr): Remove deprecated stat types warnings from items 13113 and 34967

### LINKED ISSUE #24305: worldserver - item related startup warnings
### Current Behaviour

enable warning for sql logger in config `worldserver.conf`

```
Logger.sql.sql=3,Console Errors
```

observe warnings

```
Loading Items...
Item (Entry: 13113) has deprecated stat_type3 (42)
Item (Entry: 34967) has deprecated stat_type4 (41)
Item (Entry: 34967) has deprecated stat_type5 (42)
```

### Expected Behaviour

no warnings

### Source

_No response_

### Steps to reproduce the problem

enable warning for sql logger in config `worldserver.conf`

```
Logger.sql.sql=3,Console Errors
```

observe warnings

### Extra Notes

_No response_

### AC rev. hash/commit

a3a7435ac3

### Operating system

window 10 x64

### Custom changes or Modules

_No response_

### PR DESCRIPTION
**Pull request by calm329**
## Changes Proposed:
  -  [X] Core (units, players, creatures, game systems).
  -  [ ] Scripts (bosses, spell scripts, creature scripts).
  -  [ ] Database (SAI, creatures, etc).

  ## Issues Addressed:
  - Closes #24305

  ## SOURCE:
  The changes have been validated through:
  - [X] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
    - Item 13113 (Feathermoon Headdress): Blizzard overlooked this item when converting spell damage/healing to spell power in patch 3.0. Keeping deprecated stat for 3.3.5 authenticity. See: https://www.wowhead.com/wotlk/item=13113/feathermoon-headdress#comments
    - Item 34967 (Juno's Test Gem): Test item not available to players.

  ## Tests Performed:
  - [ ] Tested in-game by the author.
  - [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
  - [X] This pull request requires further testing and may have edge cases to be tested.

  ## How to Test the Changes:
  - [X] This pull request can be tested by following the reproduction steps provided in the linked issue

  1. Enable warning for sql logger in `worldserver.conf`: `Logger.sql.sql=3,Console Errors`
  2. Start worldserver
  3. Verify no warnings appear for items 13113 and 34967

  ## Known Issues and TODO List:
  - [ ] None

---

Contribution by Gittensor, see my contribution statistics at https://gittensor.io/miners/details?githubId=148254234

### CODE REVIEW

**Comment by sogladev** [comment]
https://www.wowhead.com/wotlk/item=13113/feathermoon-headdress#comments

> Blizzard must have overlooked this when converting stat changes in patch 3.0.
>This appears to be the only item currently in game which uses the old spell power item wording, "increases damage done by magical spells and effects by up to xx", before spell damage and healing were merged into one stat, what we currently know as spell power.

For 3.3.5 accuracy, the startup error needs to be ignored for this specific item

The other 2 items look to be test items (not available to players) that also should be ignored

**Comment by calm329** [comment]
> https://www.wowhead.com/wotlk/item=13113/feathermoon-headdress#comments
> 
> > Blizzard must have overlooked this when converting stat changes in patch 3.0.
> > This appears to be the only item currently in game which uses the old spell power item wording, "increases damage done by magical spells and effects by up to xx", before spell damage and healing were merged into one stat, what we currently know as spell power.
> 
> For 3.3.5 accuracy, the startup error needs to be ignored for this specific item
> 
> The other 2 items look to be test items (not available to players) that also should be ignored

@sogladev Could you check it again?

**Comment by calm329** [comment]
> The warnings are gone now. Thanks for your PR

Thanks for your help!

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -3172,7 +3172,11 @@ void ObjectMgr::LoadItemTemplates()
             {
                 case ITEM_MOD_SPELL_HEALING_DONE:
                 case ITEM_MOD_SPELL_DAMAGE_DONE:
-                    LOG_WARN("sql.sql", "Item (Entry: {}) has deprecated stat_type{} ({})", entry, j + 1, itemTemplate.ItemStat[j].ItemStatType);
+                    // Skip warning for specific items: 13113 (Feathermoon Headdress - Blizzard oversight), 34967 (test item)
+                    if (entry != 13113 && entry != 34967)
+                    {
+                        LOG_WARN("sql.sql", "Item (Entry: {}) has deprecated stat_type{} ({})", entry, j + 1, itemTemplate.ItemStat[j].ItemStatType);
+                    }
```
Comment (sudlud) [inline src/server/game/Globals/ObjectMgr.cpp]:
```suggestion
                        LOG_WARN("sql.sql", "Item (Entry: {}) has deprecated stat_type{} ({})", entry, j + 1, itemTemplate.ItemStat[j].ItemStatType);

```

**Comment by calm329** [comment]
@sudlud I've committed your changes
